server/app/views/admin/programs/ProgramBlockAllAspectsView.java

```diff
@@ -747,4 +804,44 @@ private Modal blockDescriptionModal(
   private boolean hasNoRepeatedBlocks(ProgramDefinition programDefinition, long blockId) {
     return programDefinition.getBlockDefinitionsForEnumerator(blockId).isEmpty();
   }
+  
+  private boolean viewAllowsEditingProgram() {
+    return programDisplayType == DRAFT;
+  }
+
+  @Override
+  protected String getEditButtonText() {
+    if (viewAllowsEditingProgram()) {
+      return "Edit program details";
+    } else {
+      return "Edit Program";
+    }
+  }
+
+  @Override
+  protected String getEditButtonUrl(ProgramDefinition programDefinition) {
+    if (viewAllowsEditingProgram()) {
+      return routes.AdminProgramController.edit(programDefinition.id()).url();
+    } else {
+      try {
+        // TODO(#3162): Manually test all navigation paths once the full flow
```

Comment:
I've removed it as I'll do the testing anyways. I only put it there because I thought you'd likely have concerns about this navigation not working fully yet. I've removed it completely and return the not yet implemented error now similar to your suggestion of returning a url that is clearly wrong in the other change.